wear/src/main/java/io/homeassistant/companion/android/home/views/DetailsPanelView.kt

```diff
@@ -0,0 +1,207 @@
+import io.homeassistant.companion.android.theme.WearAppTheme
+import java.text.DateFormat
+
+const val SUPPORT_BRIGHTNESS_DEPR = 1
+val NO_BRIGHTNESS_SUPPORT = listOf("unknown", "onoff")
+const val SUPPORT_COLOR_TEMP_DEPR = 2
+const val SUPPORT_COLOR_TEMP = "color_temp"
+
+@ExperimentalComposeUiApi
+@Composable
+fun DetailsPanelView(
+    entity: Entity<*>,
+    onEntityToggled: (String, String) -> Unit,
+    onBrightnessChanged: (Float) -> Unit,
+    onColorTempChanged: (Float) -> Unit
+) {
+    WearAppTheme {
+        ThemeLazyColumn {
+            val attributes = entity.attributes as Map<*, *>
+
+            if (entity.entityId.split(".")[0] in HomePresenterImpl.toggleDomains) {
+                item {
+                    val isChecked = entity.state in listOf("on", "locked", "open", "opening")
+                    ToggleButton(
+                        checked = isChecked,
+                        onCheckedChange = { onEntityToggled(entity.entityId, entity.state) },
+                        modifier = Modifier.size(ToggleButtonDefaults.SmallToggleButtonSize)
+                    ) {
+                        ToggleChipDefaults.SwitchIcon(checked = isChecked)
+                    }
+                }
+            }
+
+            item {
+                val friendlyName = attributes["friendly_name"].toString()
+                ListHeader(friendlyName)
+            }
+
+            if (entity.entityId.split('.')[0] == "light") {
+                // Brightness
+                // On HA Core 2021.5 and later brightness detection has changed
+                // to simplify things in the app lets use both methods for now
+                val supportedColorModes = attributes["supported_color_modes"] as? List<String>
+                val supportsBrightness =
+                    if (supportedColorModes == null) false else (supportedColorModes - NO_BRIGHTNESS_SUPPORT).isNotEmpty()
+                if (supportsBrightness || ((attributes["supported_features"] as Int) and SUPPORT_BRIGHTNESS_DEPR == SUPPORT_BRIGHTNESS_DEPR)) {
+                    item {
+                        BrightnessSlider(attributes, onBrightnessChanged)
+                    }
+                }
+
+                // Color temp
+                val supportsColorTemp = supportedColorModes?.contains(SUPPORT_COLOR_TEMP) ?: false
+                val supportsColorTempComplete =
+                    supportsColorTemp || ((attributes["supported_features"] as Int) and SUPPORT_COLOR_TEMP_DEPR == SUPPORT_COLOR_TEMP_DEPR)
+                if (supportsColorTempComplete && attributes["color_mode"] == SUPPORT_COLOR_TEMP) {
+                    item {
+                        ColorTempSlider(attributes, onColorTempChanged)
+                    }
+                }
+            }
+
+            item {
+                ListHeader(R.string.details)
+            }
+            item {
+                Text(
+                    stringResource(R.string.state) + ": ${entity.state}",
```

Comment:
These `stringResource(R.string.id) + ": $value"` type of headings should probably be changed to use a string with a placeholder so the app can support languages where a different symbol is used or there has to be an empty space between the label and colon.

For example:  `<string name="last_changed">Last changed: %1$s</string>` and `stringResource(R.string.last_changed, lastChanged)`

Reply:
Should have thought of that :P

Reply:
Don't forget the brightness and color brightness sliders ;)